### PR TITLE
Do not return non-validated transaction from RippleAPI#getTransaction

### PR DESCRIPTION
**Pull request by lumberj**

### CODE REVIEW

File: src/api/ledger/transaction.js [hunk 1]
```diff
@@ -57,6 +57,11 @@ function getTransactionAsync(identifier: string, options: TransactionOptions,
 
   function callbackWrapper(error_?: Error, tx?: Object) {
     let error = error_;
+
+    if (!error && tx && tx.validated === false) {
```
Comment (clark800) [inline src/api/ledger/transaction.js]: Would be more defensive to do `tx.validated !== true`

Comment (lumberj) [inline src/api/ledger/transaction.js]: Agreed, will change

File: test/fixtures/api/rippled/tx/not-validated.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
       ],
       "TransactionIndex": 5,
       "TransactionResult": "tesSUCCESS"
-    }
+    },
+    "validated": false
```
Comment (clark800) [inline test/fixtures/api/rippled/tx/not-validated.json]: Do you know why these were missing? Does rippled ever omit "validated"?

Comment (lumberj) [inline test/fixtures/api/rippled/tx/not-validated.json]: It shouldn't

Comment (lumberj) [inline test/fixtures/api/rippled/tx/not-validated.json]: Checking

**Comment by clark800** [comment]
LGTM

**Pull request merged**